eleventy-fetch.js

```diff
@@ -34,11 +34,19 @@ function isFullUrl(url) {
 	}
 }
 
+function isAwaitable(maybeAwaitable) {
+	return (typeof maybeAwaitable === "object" && typeof maybeAwaitable.then === "function") || (maybeAwaitable.constructor.name === "AsyncFunction");
+}
+
 async function save(source, options) {
-	if(!isFullUrl(source)) {
+	if(!(isFullUrl(source) || isAwaitable(source))) {
 		return Promise.reject(new Error("Caching an already local asset is not yet supported."));
 	}
 
+	if (isAwaitable(source) && !options.formatUrlForDisplay) {
+		return Promise.reject(new Error("formatUrlForDisplay must be implemented, as a Promise has been provided."));
```

Comment:
I'm reusing the `formatUrlForDisplay` function to create a unique cache key. I'm not entirely certain this is the most intuitive name for what this is doing; maybe I should add a send option like `uniqueKey` to the options object instead?